File: PrefixSum/pivotIndex.py
from typing import List

class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        n = len(nums)
        for i in range(1,n):
            nums[i] = nums[i-1]+nums[i]
        idx = -1
        for i in range(n):
            leftVal = 0 if i==0 else nums[i-1]
            rightVal = nums[n-1]-nums[i]
            if leftVal == rightVal:
                idx = i
                break
        return idx

File: PrefixSum/PivotIndex/pivotIndex.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int n = nums.size();
        for(int i=1;i<n;i++){
            nums[i] = nums[i-1]+nums[i];
        }
        int idx = -1;
        for(int i=0;i<n;i++){
            int leftVal = i==0? 0:nums[i-1];
            int rightVal = nums[n-1]-nums[i];
            if(leftVal == rightVal){
                idx = i;
                break;
            }
        }
        return idx;
    }
};